Allow downloading of debuginfo packages from updates

AdamWill opened the pull request:
This isn't currently possible, but sometimes is useful, so add
--debuginfo to allow it.

Signed-off-by: Adam Williamson <awilliam@redhat.com>

AdamWill committed:
Allow downloading of debuginfo packages from updates

This isn't currently possible, but sometimes is useful, so add
--debuginfo to allow it.

Signed-off-by: Adam Williamson <awilliam@redhat.com>

AdamWill commented:
Note I didn't actually test this yet, relying on the CI to do it for me =)

AdamWill commented:
The test 'failures' here seem to be all bogus, AFAICT. The flake8 tests seem to be checking all the Python source in the system, or something, and flagging up tons of errors outside of bodhi entirely...

bowlofeggs commented:
Yeah we've been having huge issues with CI this week. Like no fewer than 5 separate ones. Each of them happens less than half the time, but with all their powers combined it's rare for a PR to pass tests right now, even though the tests on develop do currently pass ☹

Looks like you hit this one: https://bugs.centos.org/view.php?id=14920

bowlofeggs commented on bodhi/client/__init__.py:
```diff
@@ -675,6 +675,8 @@ def comment(update, text, karma, user, password, url, openid_api, **kwargs):
               help=('Specify arch of packages to download, "all" will retrieve packages from all '
                     'architectures'))
+@click.option('--debuginfo', is_flag=True, default=False,
+              help=('Include debuginfo packages'))
```
Let's add this to the man page - it's in ```docs/user/man_pages/bodhi.rst```

bowlofeggs commented on bodhi/client/__init__.py:
```diff
@@ -688,18 +690,21 @@ def download(url, **kwargs):
     """
     client = bindings.BodhiClient(base_url=url, staging=kwargs['staging'])
     requested_arch = kwargs['arch']
+    debuginfo = bool(kwargs['debuginfo'])
```
I think ```click``` will make this a ```bool``` for us, since it uses the ```is_flag=True``` argument above.

AdamWill replied:
I just wanted to be sure we took a copy not a reference before we delete it below.

bowlofeggs reviewed (changes requested):
Looks good - I'll take it from here and write the man page and do the bool thing for ya. Thanks for the patch!

bowlofeggs commented:
I'm going to apply this diff and rebase/force push:

```
diff --git a/bodhi/client/__init__.py b/bodhi/client/__init__.py
index b29e9a55..be3391e5 100644
--- a/bodhi/client/__init__.py
+++ b/bodhi/client/__init__.py
@@ -700,7 +700,7 @@ def download(url, **kwargs):
     """
     client = bindings.BodhiClient(base_url=url, staging=kwargs['staging'])
     requested_arch = kwargs['arch']
-    debuginfo = bool(kwargs['debuginfo'])
+    debuginfo = kwargs['debuginfo']
 
     del(kwargs['staging'])
     del(kwargs['arch'])
diff --git a/docs/user/man_pages/bodhi.rst b/docs/user/man_pages/bodhi.rst
index afed0346..939577a3 100644
--- a/docs/user/man_pages/bodhi.rst
+++ b/docs/user/man_pages/bodhi.rst
@@ -185,6 +185,10 @@ The ``updates`` command allows users to interact with bodhi updates.
     Download update(s) given by ID(s) or NVR(s). One of ``--updateid`` or
     ``builds`` is required. The download subcommand supports the following options:
 
+    ``--debuginfo``
+
+        Include debuginfo packages when downloading.
+
     ``--updateid <ids>``
 
         A comman-separated list of update IDs you would like to download.
```

bowlofeggs commented:
Oh interesting, looks like there's some breakage in our integration suite. At a glance, it looks like maybe argument order changed or something. I'll dig in and see if I can figure it out. Might just need the assertion to not care about the order of params.

bowlofeggs commented:
Heh, looks like the test assertion is def not correct. Should be easy to fix though - I'll make that a separate commit on here since it's actually not related to your change.

bowlofeggs committed:
Use a regex when asserting CLI koji calls.

The CLI download command tests did not work well for multi-build
updates because it asserted builds in the wrong order. This commit
adjusts the test so that it uses a regular expression to assert
that the right builds were downloaded so that the order of the
builds is not being asserted.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

bowlofeggs commented:
Alright, I've rebased and added a commit to hopefully fix that test assertion up.

bowlofeggs reviewed (approved):
Thanks!

bowlofeggs commented:
This patch is planned for inclusion in the upcoming 4.0.0 release: #3221

bowlofeggs commented:
Bodhi 4.0.0 beta is built and deployed to staging:

https://copr.fedorainfracloud.org/coprs/bowlofeggs/bodhi-pre-release
https://bodhi.stg.fedoraproject.org/
